web/modules/custom/asset_distribution/src/Controller/AssetDistributionController.php

```diff
@@ -48,18 +74,14 @@ public function add(RdfInterface $rdf_entity) {
    *   The access result object.
    */
   public function createAssetDistributionAccess(RdfInterface $rdf_entity) {
-    // Check that the passed in RDF entity is a solution, and that the user
-    // has the permission to create distribution entities.
-    // @todo Collection owners and facilitators should also have the right to
-    //   create distributions related to the solutions they manage.
-    // @see https://webgate.ec.europa.eu/CITnet/jira/browse/ISAICP-2450
-    if ($rdf_entity->bundle() == 'asset_release' && $this->currentUser()
-        ->hasPermission('create asset_distribution rdf entity')
-    ) {
-      return AccessResult::allowed();
+    $solution = $this->assetDistributionRelations->getReleaseSolution($rdf_entity);
+    $user = $this->currentUser();
+    if (empty($solution) && !$user->isAnonymous()) {
+      return AccessResult::neutral();
     }
```

Comment:
This check seems weird to me, can you explain? How I read it is that an asset distribution is a child item of a solution, so we actually depend on it. But this check is skipped if the user is anonymous, why is this significant? Shouldn't we always deny access if the `$solution` is missing, even if the user is anonymous?

Also, shouldn't we return `AccessResult::forbidden()` here? We are the main gatekeeper for this route, and if we can determine that the required `$solution` is missing, then we can safely decide that the route should not be accessible, right?

I always understand `AccessResult::neutral()` to mean "no opinion", or "do not have enough information to decide whether or not to grant access".